File: C++/pushing_balls.cpp
// hey, i didn't choose the name
#include <bits/stdc++.h>
using namespace std;

#ifdef LOCAL
#include "debug.h"
#else
#define debug(...)
#endif

int main(){
    int c; cin >> c;
    while (c--) {
        int n, m; cin >> n >> m;
        vector<string> grid;
        for (int i = 0; i < n; i++){
            string tmp; cin >> tmp;
            grid.push_back(tmp);
        }
        debug(grid);
        bool valid = true;
        for (int i = 1; i < n; i++){
            for (int j = 1; j < m; j++){
                debug(grid[i][j]);
                debug(i);
                debug(j);
                if (grid[i][j] == '0') continue;
                if (grid[i-1][j] == '0' and grid[i][j-1] == '0'){
                    debug("end");
                    valid = false;
                    break;
                }
            }
            if (not valid) break;
        }

        valid ? puts("YES") : puts("NO");
    }

}

File: python/pushing_balls.py
#!/bin/pypy3
for c in range(int(input())):
    l, w = map(int, input().split())
    grid = [input().strip() for _ in range(l)]
    valid = True
    for i in range(1, l):
        for j in range(1, w):
            if (grid[i-1][j] == '0' and grid[i][j-1] == '0' and grid[i][j] == '1'):
                #print(f"{i=} {j=}")
                #print(f"{grid=}")
                valid = False
                break
        if (not valid): break

    print("YES") if valid else print("NO")

    
